libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -314,6 +355,154 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
+{
+	ASTString const& member = _memberAccess.memberName();
+	if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type))
+		if (funType->bound())
+		{
+			solUnimplementedAssert(false, "");
+		}
+
+	switch (_memberAccess.expression().annotation().type->category())
+	{
+	case Type::Category::Contract:
+	{
+		ContractType const& type = dynamic_cast<ContractType const&>(*_memberAccess.expression().annotation().type);
+		if (type.isSuper())
+		{
+			solUnimplementedAssert(false, "");
+		}
+		// ordinary contract type
+		else if (Declaration const* declaration = _memberAccess.annotation().referencedDeclaration)
+		{
+			u256 identifier;
+			if (auto const* variable = dynamic_cast<VariableDeclaration const*>(declaration))
+				identifier = FunctionType(*variable).externalIdentifier();
+			else if (auto const* function = dynamic_cast<FunctionDefinition const*>(declaration))
+				identifier = FunctionType(*function).externalIdentifier();
+			else
+				solAssert(false, "Contract member is neither variable nor function.");
+			// TODO here, we need to assign address and function identifier to two variables.
+			// We migt also just combine them into a single variable already....
+			solUnimplementedAssert(false, "");
+		}
+		else
+			solAssert(false, "Invalid member access in contract");
+		break;
+	}
+	case Type::Category::Integer:
+	{
+		solAssert(false, "Invalid member access to integer");
+		break;
+	}
+	case Type::Category::Address:
+	{
+		if (member == "balance")
+			defineExpression(_memberAccess) <<
+				"balance(" <<
+				expressionAsType(_memberAccess.expression(), *TypeProvider::address()) <<
+				")\n";
+		else if ((set<string>{"send", "transfer"}).count(member))
+		{
+			solAssert(dynamic_cast<AddressType const&>(*_memberAccess.expression().annotation().type).stateMutability() == StateMutability::Payable, "");
+			defineExpression(_memberAccess) <<
+				expressionAsType(_memberAccess.expression(), *TypeProvider::payableAddress()) <<
```

Comment:
Is this incomplete?

Reply:
It's fine - it is just the function `x.send`, not the function call `x.send()`.